[New] `no-unused-modules`: Add `ignoreUnusedTypeExports` option (#3011)
import-js/eslint-plugin-import: docs/rules/no-unused-modules.md

 - **`missingExports`**: if `true`, files without any exports are reported (defaults to `false`)
 - **`unusedExports`**: if `true`, exports without any static usage within other modules are reported (defaults to `false`)

>  - **`ignoreUnusedTypeExports`**: if `true`, type exports without any static usage within other modules are reported (defaults to `false` and depends on `unusedExports` being `true`)

Author: ```suggestion
 - **`ignoreUnusedTypeExports`**: if `true`, type exports without any static usage within other modules are reported (defaults to `false` and has no effect unless `unusedExports` is `true`)
```
